src/library/scala/collection/convert/impl/IteratorStepper.scala

```diff
@@ -119,9 +120,9 @@ private[collection] class LongIteratorStepper(_underlying: Iterator[Long])
 }
 
 /** Common functionality for Steppers that step through an Iterator, caching the results as needed when a split is requested. */
-private[convert] abstract class IteratorStepperBase[A, SP >: Null <: Stepper[A], Semi <: SP](final protected var underlying: Iterator[A]) {
+private[convert] abstract class IteratorStepperBase[+A, +SP >: Null <: Stepper[A], +Semi <: SP](final protected var underlying: Iterator[A @uncheckedVariance]) {
```

Comment:
I'm uneasy about the `@uncheckedVariance` here.  Maybe `underlying` is always used in such a way that it doesn't admit misuse (e.g. assigning a wider type in a context where you don't see the narrowest `A`, and then pulling these things out when you don't know how to handle them).  But it's probably worth a little thought about whether it's possible to exploit this to break things.  (I don't have time right now to think about it in enough detail.)

Reply:
Actually, `underlying` doesn't need to be a `var`, I pushed that change here. But the `proxied` field below is still marked `uncheckedVariance`. I looked how it's used, and it seems safe, but I might very well have missed something.